base/linalg/dense.jl

```diff
@@ -673,11 +673,9 @@ function inv(A::StridedMatrix{T}) where T
     S = typeof((one(T)*zero(T) + one(T)*zero(T))/one(T))
     AA = convert(AbstractArray{S}, A)
     if istriu(AA)
-        Ai = inv(UpperTriangular(AA))
-        Ai = convert(typeof(parent(Ai)), Ai)
+        Ai = triu!(parent(inv(UpperTriangular(AA))))
```

Comment:
You could have used `full!` here. As mentioned in https://github.com/JuliaLang/julia/pull/24137#issuecomment-336612197, it would be better to use a generic function for this operation.

Reply:
Agreed. Full response in the other thread :).